ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/backwardsync/BackwardSyncStep.java

```diff
@@ -89,13 +99,50 @@ protected Hash earliestUnprocessedHash() {
             });
   }
 
+  @VisibleForTesting
+  protected CompletableFuture<List<BlockHeader>> requestHeaders(final Hash hash) {
+    debugLambda(LOG, "Requesting header for hash {}", () -> hash.toString().substring(0, 20));
+    return GetHeadersFromPeerByHashTask.endingAtHash(
+            context.getProtocolSchedule(),
+            context.getEthContext(),
+            hash,
+            context.getProtocolContext().getBlockchain().getChainHead().getHeight(),
+            BackwardsSyncContext.BATCH_SIZE,
+            context.getMetricsSystem())
+        .run()
+        .thenApply(
+            peerResult -> {
+              final List<BlockHeader> result = peerResult.getResult();
+              if (result.isEmpty()) {
+                throw new BackwardSyncException(
+                    "Did not receive a header for hash {}" + hash.toString().substring(0, 20));
+              }
+              infoLambda(
+                  LOG,
+                  "Got headers {} from height {} to {} height {}",
```

Comment:
I like this format that you used in another spot:
        "Got headers from {} (height {}) to {} (height {})",